verilog/CST/functions.cc

```diff
@@ -125,10 +178,43 @@ const verible::SyntaxTreeNode* GetLocalRootFromFunctionCall(
 
 const verible::SyntaxTreeNode* GetIdentifiersFromFunctionCall(
     const verible::Symbol& function_call) {
-  const verible::SyntaxTreeNode* local_root = GetSubtreeAsNode(
-      function_call, NodeEnum::kFunctionCall, 0, NodeEnum::kLocalRoot);
-  if (!local_root) return nullptr;
-  const verible::Symbol* identifier = GetIdentifiersFromLocalRoot(*local_root);
+  const verible::Symbol* reference = nullptr;
+  const verible::Symbol* reference_call_base = nullptr;
+  const verible::Symbol* identifier = nullptr;
+  if (function_call.Tag().tag == (int)NodeEnum::kFunctionCall) {
+    reference_call_base =
+        GetSubtreeAsSymbol(function_call, NodeEnum::kFunctionCall, 0);
+    if (reference_call_base->Tag().tag != (int)NodeEnum::kReferenceCallBase) {
+      return nullptr;
+    }
+    reference = GetSubtreeAsSymbol(*reference_call_base,
```

Comment:
Would this logic be simplified if the some of these subtrees were transformed into proper function call syntax nodes?

Reply:
I have simplified it about as much as I could without removing some sanity checks (like all the `... != nullptr` checking and NodeEnum equalities)